hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/common/AutoCloseDataSetLock.java

```diff
@@ -30,17 +33,39 @@
  * use a try-with-resource syntax.
  */
 public class AutoCloseDataSetLock extends AutoCloseableLock {
-  private Lock lock;
+
+  private final Lock lock;
+
+  private final DataNodeLockManager<AutoCloseDataSetLock> dataNodeLockManager;
+
+  private final boolean isReadLock;
+
+  private static ThreadLocal<DataSetLockHeldInfo> heldInfoThreadLocal = new
```

Comment:
Thank you for your pr. It's necessary to add metrics, but there's an issue that needs clarification. In certain versions of JDK 17, particularly with G1, there are problems with ThreadLocal where entries cannot be properly GC'd. This issue can cause unusually high CPU usage. I'll provide more details later.

Reply:
Sorry for the delayed response. I've been quite busy recently.
Here's the detailed situation:

In certain versions of JDK 17, there is an issue where weak references to ThreadLocal cannot be reclaimed(There is a problem with G1, but CMS can reclaim memory normally). This results in ThreadLocal objects accumulating a large amount of data, causing significant CPU overhead during traversal and element retrieval. We initially noticed this issue during latency profiling related to Ozone OM locks, where CPU usage for our OM spiked over a period of time. Hence, I suggested that enabling lock profiling should ideally be an optional feature.

Reply:
> Sorry for the delayed response. I've been quite busy recently. Here's the detailed situation:
> 
> In certain versions of JDK 17, there is an issue where weak references to ThreadLocal cannot be reclaimed(There is a problem with G1, but CMS can reclaim memory normally). This results in ThreadLocal objects accumulating a large amount of data, causing significant CPU overhead during traversal and element retrieval. We initially noticed this issue during latency profiling related to Ozone OM locks, where CPU usage for our OM spiked over a period of time. Hence, I suggested that enabling lock profiling should ideally be an optional feature.
> 
> OpenJDK 17 lower version bug: https://bugs.openjdk.org/browse/JDK-8188055.
> 
> We need to make sure that the JDK17 version we choose has the above patch.

First of all, I'm very sorry that I haven't been able to reply to you recently because I've been busy with work. Secondly, thank you very much for your suggestion. I will modify this function to be optional later. @slfan1989